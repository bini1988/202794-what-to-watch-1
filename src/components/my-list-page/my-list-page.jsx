import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {Operation} from "../../reducer/movies/movies";
import {getMyListMovies} from "../../reducer/movies/selectors";

import PageHeader from "../page-header/page-header";
import PageTitle from "../page-title/page-title";
import PageFooter from "../page-footer/page-footer";
import UserBlock from "../user-block/user-block";
import MoviesCatalog from "../movies-catalog/movies-catalog";

class MyListPage extends PureComponent {
  render() {
    const {movies} = this.props;

    return (
      <div className="user-page">
        <PageHeader className="user-page__head">
          <PageTitle className="user-page__title">
            {`My list`}
          </PageTitle>
          <UserBlock/>
        </PageHeader>
        <MoviesCatalog
          movies={movies}/>
        <PageFooter/>
      </div>
    );
  }

  componentDidMount() {
    this.props.onMyListMoviesFetch();
  }
}

MyListPage.propTypes = {
  /** Список фильмов добавленных в список «к просмотру» */
  movies: PropTypes.arrayOf(
      PropTypes.object,
  ),
  /** Получить список фильмов добавленных в список «к просмотру» */
  onMyListMoviesFetch: PropTypes.func,
};

const mapStateToProps = (state) => {
  return {
    movies: getMyListMovies(state),
  };
};
const mapDispatchToProps = {
  onMyListMoviesFetch: Operation.fetchMyListMovies,
};

export {MyListPage};
export default connect(mapStateToProps, mapDispatchToProps)(MyListPage);

