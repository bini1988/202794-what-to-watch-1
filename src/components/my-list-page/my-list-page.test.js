import React from "react";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import genreGroupsMock from "../../mocks/movies-groups";
import {MyListPage} from "./my-list-page";

it(`MyListPage correctly renders default markup`, () => {
  const tree = renderer.create(
      <MemoryRouter
        initialEntries = {[`/`]}>
        <MyListPage
          moviesGenreGroups={genreGroupsMock}
          myListMovies={[]}
          fetchMovies={() => {}}
          fetchMyListMovies={() => {}}
        />
      </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
