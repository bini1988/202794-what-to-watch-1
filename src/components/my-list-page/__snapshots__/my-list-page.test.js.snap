// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListPage correctly renders default markup 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        id="movie-1"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/film/1"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              onCanPlayThrough={[Function]}
              onEnded={[Function]}
              onLoadStart={[Function]}
              onPause={[Function]}
              onTimeUpdate={[Function]}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/midnight-special.jpg"
              preload="none"
              src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            Midnight Special
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        id="movie-2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/film/2"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              onCanPlayThrough={[Function]}
              onEnded={[Function]}
              onLoadStart={[Function]}
              onPause={[Function]}
              onTimeUpdate={[Function]}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/pulp-fiction.jpg"
              preload="none"
              src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            Pulp Fiction
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        id="movie-3"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/film/3"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              onCanPlayThrough={[Function]}
              onEnded={[Function]}
              onLoadStart={[Function]}
              onPause={[Function]}
              onTimeUpdate={[Function]}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
              preload="none"
              src="http://media.xiph.org/mango/tears_of_steel_1080p.webm"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            Bronson
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        id="movie-4"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/film/4"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              onCanPlayThrough={[Function]}
              onEnded={[Function]}
              onLoadStart={[Function]}
              onPause={[Function]}
              onTimeUpdate={[Function]}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
              preload="none"
              src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            What We Do in the Shadows
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        id="movie-5"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/film/5"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              onCanPlayThrough={[Function]}
              onEnded={[Function]}
              onLoadStart={[Function]}
              onPause={[Function]}
              onTimeUpdate={[Function]}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              preload="none"
              src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h3>
        </a>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
